### DISCUSSION TITLE
SQL modification queries allowed while in read only mode (when auto commit enabled)

### DISCUSSION THREAD

**Discussion by burhomer** [dbeaver/dbeaver#11970] [Q&A]
I reproduced this issue in DBeaver 7.3.2.202101032114 on Windows 10.

Drivers: PostgreSQL JDBC 

I create a new connection and I set:
Security: Read only connection
Connection type: read-only where the connection type is:
![ro](https://user-images.githubusercontent.com/3050999/104307880-16f4ba00-54d0-11eb-8190-5a482fd478fb.png)

If I try to run modification data or schema is the same I get the message I expect:
SQL Error [25006]: ERROR: cannot execute DROP TABLE in a read-only transaction

DBeaver also doesn't allow me to to edit the database using the GUI.

If I change the connection type enabling auto-commit (NB: bouncing the connection is needed):
![read  only](https://user-images.githubusercontent.com/3050999/104308341-c92c8180-54d0-11eb-9798-8f9bc96b28c6.png)
the behavour changes completely.

While the GUI keeps preventing DB changes to the DB, I'm now be able to modify the DB using SQL (for example dropping a table using SQL) without being notified.

**Comment by kseniiaguzeeva** [comment]
I cannot reproduce it. What do we do differently?
My steps:

1. Select a type where auto-commit is off, set read-only for the connection
2. Press Test connection and Ok
3. Try to execute Drop query in SQL Editor > the action is not allowed
4. Go back to the connection settings and enable auto-commit for the connection type
5. Press Test connection and Ok
6. Try to execute Drop query in SQL Editor > **the action is not allowed** again

https://user-images.githubusercontent.com/51405061/104912493-638f3800-599d-11eb-9833-1ee10f29869f.mp4

**Comment by burhomer** [comment]
Hi,

I made exactly the same steps. Today I tried to:
- remove DBeaver with its DBeaverData folder
- install the latest version of DBeaver Community (without importing old DBeaver settings)

I still have this issue

I wonder if this could be caused by the version of the drivers. I had to upgrade to the latest a few months ago:
![drivers](https://user-images.githubusercontent.com/3050999/105184835-18f1f500-5b30-11eb-8012-320fc90f780f.png)

Simone

**Comment by burhomer** [comment]
I confirm the problem is with the drivers.

By selecting older drivers the behaviour is correct. This is the maximum version I can use without having the issue.

![middle](https://user-images.githubusercontent.com/3050999/105191410-5443f200-5b37-11eb-85a4-f21a2040aa30.png)

This version also allows me to access my DB (version 10)

**Comment by kseniiaguzeeva** [comment]
Thank you for the additional info. I can reproduce it now. Will be fixed in future releases.

**Comment by serge-rider** [comment]
I'm not sure that we can fix it.
You need to turn on option "Read-only connection" to restrict data modification on server side. 
Because it is not possible to do this on client-side. Because any stored procedure or function may also do data modifications and only server can check it.

"Edit permissions" restricts explicit data/metadata modification actions on client only.
